packages/legend-query-builder/src/stores/QueryBuilderState.ts

```diff
@@ -378,6 +378,7 @@ export abstract class QueryBuilderState implements CommandRegistrar {
 
   initializeWithQuery(query: RawLambda): void {
     this.rebuildWithQuery(query);
+    this.resetQueryResult();
```

Comment:
let's modify `resetQueryResult()` to take advanced options to preserve the result state `options?: { preserveResult }`